Sources/StreamChat/Workers/CurrentUserUpdater_Tests.swift

```diff
@@ -494,31 +496,41 @@ final class CurrentUserUpdater_Tests: StressTestCase {
         try database.writeSynchronously {
             try $0.saveCurrentUser(payload: userPayload)
         }
-        
+
+        // Assert data is stored in the DB
+        var currentUser: CurrentChatUser? {
+            database.viewContext.currentUser()?.asModel()
+        }
+
+        // Make sure no devices are stored in the DB
+        assert(currentUser?.devices.isEmpty == true)
+
         // Save previous device to the db
         try database.writeSynchronously {
-            try $0.saveCurrentUserDevices([.dummy])
+            // Simulate 4 devices exist in the DB
+            try $0.saveCurrentUserDevices([.dummy, .dummy, .dummy, .dummy])
         }
-        
+
         // Call updateDevices
+        var callbackCalled = false
         currentUserUpdater.updateDevices(currentUserId: .unique) {
             // No error should be returned
             XCTAssertNil($0)
+            callbackCalled = true
         }
         
         // Simulate API response with devices data
         let dummyDevices = DeviceListPayload.dummy
+        assert(dummyDevices.devices.isEmpty == false)
         apiClient.test_simulateResponse(.success(dummyDevices))
-        
-        // Assert data is stored in the DB
-        var currentUser: CurrentChatUser? {
-            database.viewContext.currentUser()?.asModel()
-        }
-        
+
         // Previous devices should not be cleared
         AssertAsync {
-            Assert.willBeEqual(currentUser?.devices.count, 2)
-            Assert.willBeEqual(currentUser?.devices.last?.id, dummyDevices.devices.last?.id)
+            Assert.willBeEqual(
+                currentUser?.devices.map(\.id).sorted(),
+                dummyDevices.devices.map(\.id).sorted()
+            )
+            Assert.willBeTrue(callbackCalled)
         }
```

Comment:
This test actually meant to verify that if we had previous devices already added, they were not deleted, so I stored 1 device first, and then called updateDevices, and it should not clear them. Because the request of update devices was returning empty array. So if we have stored devices, this will clear all devices.